src/components/record/RecordForm.js

```diff
@@ -119,60 +142,88 @@ function AttachmentInfo(props: AttachmentInfoProps) {
   if (!attachment) {
     return null;
   }
+
   return (
     <div className="panel panel-default attachment-info">
       <div className="panel-heading">
         <i className="glyphicon glyphicon-paperclip" />
         <b>Attachment information</b>
       </div>
       <div className="panel-body">
-        {attachmentRequired && (
-          <div className="alert alert-warning">
-            <p>
-              An attachment is required for records in this collection. To
-              replace current attachment, use the <i>File attachment</i> field
-              below.
-            </p>
-          </div>
-        )}
-        <AttachmentPreview
-          mimetype={attachment.mimetype}
-          location={buildAttachmentUrl(record, capabilities)}
-        />
-        <table className="table table-condensed">
-          <tbody>
-            <tr>
-              <th>Location</th>
-              <td>{linkify(attachment.location)}</td>
-            </tr>
-            <tr>
-              <th>Filename</th>
-              <td>{attachment.filename}</td>
-            </tr>
-            <tr>
-              <th>Size</th>
-              <td>{filesize(attachment.size)}</td>
-            </tr>
-            <tr>
-              <th>Hash</th>
-              <td>{attachment.hash}</td>
-            </tr>
-            <tr>
-              <th>Mime-Type</th>
-              <td>{attachment.mimetype}</td>
-            </tr>
-          </tbody>
-        </table>
-        {!attachmentRequired && (
-          <p className="text-right attachment-action">
-            <input
-              type="button"
-              onClick={deleteAttachment}
-              className="btn btn-danger"
-              value="Delete this attachment"
+        {allowEditing &&
+          attachmentRequired && (
+            <div className="alert alert-warning">
+              <p>
+                An attachment is required for records in this collection. To
+                replace current attachment, use the <i>File attachment</i> field
+                below.
+              </p>
+            </div>
+          )}
+        <div className="attachment-attributes">
+          <table className="table table-condensed">
+            <tbody>
+              <tr>
+                <th>Location</th>
+                <td>{linkify(attachment.location)}</td>
+              </tr>
+              <tr>
+                <th>Filename</th>
+                <td>{attachment.filename}</td>
+              </tr>
+              <tr>
+                <th>Size</th>
+                <td>{filesize(attachment.size)}</td>
+              </tr>
+              <tr>
+                <th>Hash</th>
+                <td>{attachment.hash}</td>
+              </tr>
+              <tr>
+                <th>Mime-Type</th>
+                <td>{attachment.mimetype}</td>
+              </tr>
+            </tbody>
+          </table>
+          {!!attachment.original && (
```

Comment:
```
> kinto-admin@1.20.1 flow-check /home/mathieu/Code/Mozilla/kinto-admin
> flow check

Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ src/components/record/RecordForm.js:202:44

Cannot get attachment.original.hash because property hash is missing in undefined [1].

     src/components/record/RecordForm.js
     199│                   <td>{filesize(attachment.original.size)}</td>
     200│                 </tr>
     201│                 <tr>
     202│                   <td>{attachment.original.hash}</td>
     203│                 </tr>
     204│                 <tr>
     205│                   <td>{attachment.original.mimetype}</td>

     src/types.js
 [1]  27│   original?: {
      28│     filename: string,
      29│     size: number,
      30│     hash: string,
      31│     mimetype: string,
      32│   },

```
How I can tell Flow that this is safe??

Reply:
Shouldn't this be `attachment.original &&`, not `!attachment.original &&`?

Reply:
Oops, I misread it, it's a double `!`..